metricflow/constraints/time_constraint.py

```diff
@@ -17,6 +17,20 @@ class TimeRangeConstraint(SerializableDataclass):
     start_time: datetime.datetime
     end_time: datetime.datetime
 
+    def __post_init__(self) -> None:  # noqa: D
+        if self.start_time > self.end_time:
+            raise RuntimeError(
+                f"start_time must not be > end_time. start_time={self.start_time} end_time={self.end_time}"
+            )
+
+        if self.start_time < TimeRangeConstraint.ALL_TIME_BEGIN():
+            raise RuntimeError(
+                f"start_time={self.start_time} exceeds the limits of {TimeRangeConstraint.ALL_TIME_BEGIN()}"
+            )
+
+        if self.end_time > TimeRangeConstraint.ALL_TIME_END():
+            raise RuntimeError(f"end_time={self.end_time} exceeds the limits of {TimeRangeConstraint.ALL_TIME_END()}")
```

Comment:
2040 isn't too far away but I guess that's alright.

Out of historical instance, why is this here? And why is it a fixed constant? Shouldn't we make it slide forward based on current date?

Reply:
So the limits of the `TimeRangeConstraint` are used for the time spine table. We could increase this, but that would cause the size of the time spine table to be larger, which adds a bit of overhead when creating it.

Generally prefer not to make it sliding to get better consistency + easier debugging.

Reply:
> the limits of the TimeRangeConstraint are used for the time spine table

Ah, makes sense!